### PR TITLE
feat(html): matching labels with dash&spec symbol

### PR DESCRIPTION
**Pull request by alexneo2003**
related to #22700 

https://user-images.githubusercontent.com/11705902/235174845-bceaaa07-7ea2-4c9f-839b-041bdd21cd6d.mp4

### CODE REVIEW

File: packages/html-reporter/src/labelUtils.tsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ export function escapeRegExp(string: string) {
 
 // match all tags in test title
 export function matchTags(title: string): string[] {
-  return title.match(/@(\w+)/g)?.map(tag => tag.slice(1)) || [];
+  return title.match(/@([\w-]+)/g)?.map(tag => tag.slice(1)) || [];
```
Comment (pavelfeldman) [inline packages/html-reporter/src/labelUtils.tsx:28]: what about underscore?

Comment (alexneo2003) [reply packages/html-reporter/src/labelUtils.tsx:28]: `\w` - matches any word characters, including underscore `_`
`-` - is exceptional character what using somebody for tags
`# $ % ^ & * + = < > |` - are more special symbols, and i think what almost nobody not using it for tags
If tag will containing special symbol it should be escaped by search engine and no longer work

Comment (pavelfeldman) [reply packages/html-reporter/src/labelUtils.tsx:28]: What is the conceptual difference between `-` and `+` when it comes to the tab value, why do you want to treat them differently?

Comment (alexneo2003) [reply packages/html-reporter/src/labelUtils.tsx:28]: `[\w-]+` - it means
`\w` - any one of [a-zA-Z0-9_]
`-` - one of `-`
`+` - as much as possible
It match like `@CBC-123`, `@flaky`
Or I'm not understand you what you want)
I offered such a solution to the "problem", you have a different vision of it.  we still won't be able to cover all possible options) but we should try to do so.  and it is necessary to exclude options that, on the contrary, can harm.

Comment (pavelfeldman) [reply packages/html-reporter/src/labelUtils.tsx:28]: I am suggesting to be more lax in matching the tags. Clearly we have already underestimated the expressiveness of the tags, hence this follow up. So no considerations like "more special" stand at this point. I'm thinking about tags like:

`@smoke:p1`
`@issue[123]`
`@issue#123`
`@$$$`
`@tl/dr`

etc.

Hence my question of why not treat any uninterrupted sequence as a tag and minimize the number of follow up patches like this one.

Comment (pavelfeldman) [reply packages/html-reporter/src/labelUtils.tsx:28]: (backslash escaping space-separated tags would also be fine)

Comment (alexneo2003) [reply packages/html-reporter/src/labelUtils.tsx]: i'm try using `escapeRegExp` when matching tags from test title but after they looks like `@issue\[123\]`, `@\$\$\$` and after be clicked this string not matching anything


https://user-images.githubusercontent.com/11705902/235366845-6139f600-8045-41f6-9c8b-72d6ea68a96a.mp4

**Review by pavelfeldman — APPROVED** [review]

**Pull request merged**